scripts/repoint_frontend.py

```diff
@@ -0,0 +1,198 @@
+    temp = []
+    for line in lines_to_add:
+        try:
+            temp.append(line.format(shortcode=shortcode))
+        except (ValueError, KeyError):
+            # this function trips up on  the {% %} syntax, dirty workaround is to add those lines without formatting
+            temp.append(line)
+
+    lines_to_add = temp
+
+    block_found = False
+    new_lines = []
+    i = 0
+
+    while i < len(lines):
+        # add lines to the new content
+        new_lines.append(lines[i])
+
+        # check for the block start
+        if lines[i].strip() == block_to_follow[0]:
+            block_match = True
+            # check if the subsequent lines match the rest of the block
+            for j in range(1, len(block_to_follow)):
+                if i + j >= len(lines):
+                    block_match = False
+                    break
+
+                # add inspected lines to the new content
+                new_lines.append(lines[i + j])
+
+                if lines[i + j].strip() != block_to_follow[j]:
+                    block_match = False
+                    break
+
+            # if the block matches fully add lines after the block
+            if block_match:
+                block_found = True
+                # move the index to the end of the block
+                i += len(block_to_follow)
+                # add the lines after the block
+                new_lines.extend([f"{line}\n" for line in lines_to_add])
+                continue  # skip to next iteration after inserting lines
+
+        i += 1
+
+    if not block_found:  # noqa: E713
+        print(f"Error: Block not found in the file at {file_path}.")
+        return
+
+    with open(file_path, 'w') as file:
+        file.writelines(new_lines)
+
+    print(f"Lines successfully added after the specified block in {file_path}.")
+
+
+if __name__ == "__main__":
+    current_directory = os.getcwd()
+    directory_name = os.path.basename(current_directory)
+
+    if (directory_name != "communications-manager-api"):
+        print("Error: repoint_frontend must be run from the project root. Current dir:", directory_name)
+        sys.exit(1)
+
+    parser = argparse.ArgumentParser(description="Script to modify the target.xml file based on the environment argument.")  # noqa: E501
+
+    parser.add_argument('shortcode', type=str, help="Environment identifier (e.g. 'gith1')")
+    args = parser.parse_args()
+
+    # modify target.xml
+    file_path = 'proxies/live/apiproxy/targets/target.xml'
+    modify_target_file(args.shortcode, file_path)
+
+    # modify AssignMessage.MessageBatches.Create.Request.xml
+    file_path = 'proxies/shared/policies/AssignMessage.MessageBatches.Create.Request.xml'
+    block_to_follow = [
+        '<AssignVariable>',
+        '<Name>requestpath</Name>',
+        '<Value>/api/v1/send</Value>',
+        '</AssignVariable>'
+    ]
+    lines_to_add = [
+        '    {{% if ENVIRONMENT_TYPE != \'sandbox\' %}}',
```

Comment:
Could you not encode the line endings and just have this as a string instead of an array to make the file manipulation more straightforward?

Reply:
I think that that would make this code less readable and maintainable

Reply:
Just noticed that the line adding code is on one line so no notable difference :)